webextension/js/popup.js

```diff
@@ -1,150 +1,335 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
 /* global browser, window, document, localStorage */
+
 const CONTAINER_HIDE_SRC = "/img/container-hide.svg";
 const CONTAINER_UNHIDE_SRC = "/img/container-unhide.svg";
 
-function showPanel(panelSelector) {
-  for (let panelElement of document.querySelectorAll(".panel")) {
-    panelElement.classList.add("hide");
+// Let's set it to false before releasing!!!
+const DEBUG = false;
+
+// List of panels
+const P_ONBOARDING_1     = "onboarding1";
+const P_ONBOARDING_2     = "onboarding2";
+const P_CONTAINERS_LIST  = "containersList";
+const P_CONTAINERS_EDIT  = "containersEdit";
+const P_CONTAINER_INFO   = "containerInfo";
+const P_CONTAINER_EDIT   = "containerEdit";
+const P_CONTAINER_DELETE = "containerDelete";
+
+function log(...args) {
+  if (DEBUG) {
+    console.log.call(console, ...args);
   }
-  document.querySelector(panelSelector).classList.remove("hide");
 }
 
-function showContainerTabsPanel(identity) {
-  // Populating the panel: name and icon
-  document.getElementById("container-info-name").innerText = identity.name;
+// This object controls all the panels, identities and many other things.
+let Logic = {
+  _identities: [],
+  _currentIdentity: null,
+  _currentPanel: null,
+  _previousPanel: null,
+  _panels: {},
+
+  init() {
+    // Retrieve the list of identities.
+    this.refreshIdentities()
+
+    // Routing to the correct panel.
+    .then(() => {
+      if (localStorage.getItem("onboarded2")) {
+        this.showPanel(P_CONTAINERS_LIST);
+      } else if (localStorage.getItem("onboarded1")) {
+        this.showPanel(P_ONBOARDING_2);
+      } else {
+        this.showPanel(P_ONBOARDING_1);
+      }
+    });
+  },
 
-  let icon = document.getElementById("container-info-icon");
-  icon.setAttribute("data-identity-icon", identity.image);
-  icon.setAttribute("data-identity-color", identity.color);
+  refreshIdentities() {
+    return browser.runtime.sendMessage({
+      method: "queryIdentities"
+    })
+    .then(identities => {
+      this._identities = identities;
+    });
+  },
 
-  // Show or not the has-tabs section.
-  for (let trHasTabs of document.getElementsByClassName("container-info-has-tabs")) {
-    trHasTabs.hidden = !identity.hasHiddenTabs && !identity.hasOpenTabs;
-    trHasTabs.setAttribute("data-user-context-id", identity.userContextId);
-  }
+  showPanel(panel, currentIdentity = null) {
+    // Invalid panel... ?!?
+    if (!(panel in this._panels)) {
+      throw("Something really bad happened. Unknown panel: " + panel + "\n");
```

Comment:
Re: http://eslint.org/docs/rules/no-throw-literal

Do we want to throw strings or Error objects?